From f763e48755c9f7db5e4beecc39b1426eb951df9a Mon Sep 17 00:00:00 2001
From: Pavel Nedev <pnedev@mm-sol.com>
Date: Fri, 28 Jan 2011 13:53:04 +0200
Subject: OMAP4/S3D V4L2: Change rotation direction

Rotation was applied counter-clockwise -> say 90 degrees corresponded to 270 degrees
of the 2D V4L2 driver (omap_vout).
This alignes both 2D and 3D V4L2 drivers rotation directions.

Change-Id: Ie50511e2162740713e2a6d1074a2e8a706a6ecef
Signed-off-by: Pavel Nedev <pnedev@mm-sol.com>
---
 drivers/media/video/omap/omap_s3d_overlay.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/video/omap/omap_s3d_overlay.c b/drivers/media/video/omap/omap_s3d_overlay.c
index 6813b10..8fb0089 100644
--- a/drivers/media/video/omap/omap_s3d_overlay.c
+++ b/drivers/media/video/omap/omap_s3d_overlay.c
@@ -265,13 +265,13 @@ static int v4l2_rot_to_dss_rot(const unsigned int v4l2_rotation,
 {
 	switch (v4l2_rotation) {
 	case 90:
-		*rotation = OMAP_DSS_ROT_270;
+		*rotation = OMAP_DSS_ROT_90;
 		break;
 	case 180:
 		*rotation = OMAP_DSS_ROT_180;
 		break;
 	case 270:
-		*rotation = OMAP_DSS_ROT_90;
+		*rotation = OMAP_DSS_ROT_270;
 		break;
 	case 0:
 		*rotation = OMAP_DSS_ROT_0;
-- 
1.6.0.4

